### ISSUE TITLE
How make cert-ignore?

### ISSUE THREAD

**Issue by dronmaxman** [rdesktop/rdesktop#349]
I have use rdesktop 1.9.0 with vmware horizon. And I can't press YES.
What key help me to skip server certificate cheack?

![Screen Shot 2019-09-28 at 02 28 30](https://user-images.githubusercontent.com/15871669/65807641-154a9a00-e198-11e9-8a9a-6695008a4e8a.jpg)

**Comment by CendioOssman** [comment]
I'm afraid we have not added a command line argument to ignore bad certificates. So right now you need to run rdesktop at least once to approve that certificate, or make sure the certificate is otherwise trusted by the system.

**Comment by offsides** [comment]
Can I also request a command-line switch to auto-accept certs?  I use multiple SSH tunnels to bounce my rdesktop connections, and have a script to do it all for me.  Having to manually establish the tunnels just to be able to accept the certificate from a known system (just because it auto-renews every 6 months, if not more often) is pretty frustrating.  I understand why you don't want to auto-accept by default, but not even having the option is unfortunate.

**Comment by nathanshearer** [comment]
I would also like to request an option similar to the one that wget uses:
--no-check-certificate

For now, I have been piping "yes" into rdesktop, but I don't like blindly answering yes to whatever it is asking me for all my connections.

**Comment by MarkLTZ** [comment]
Hello,
PR #364 should address this issue. Could someone check and merge it on master repo?

**Comment by petersun1977** [comment]
> 
> 
> I'm afraid we have not added a command line argument to ignore bad certificates. So right now you need to run rdesktop at least once to approve that certificate, or make sure the certificate is otherwise trusted by the system.

I can accept type "rdesktop RDS_SERVER" every time the cert is changed(for us it is every 6 months), but doing this only adds a new cert in /home/USER/.local/share/rdesktop/certs/known_certs, but does not remove the old cert, so the user has to remove old one manually.

**Comment by MarkLTZ** [comment]
@petersun1977 did you check my PR? #364

**Comment by twoninenine** [comment]
@dronmaxman if this is still relevant for you, I fount out following solution
`echo 'yes' | rdesktop <ip>`

**Comment by jctmf** [comment]
Need it too.

**Comment by HomeBoy38** [comment]
Ignore certificate is definitely an option I will hope to see some day.

I tried the echo yes solution, unfortunately, rdesktop crashes with 1.9.0 version if the password is provided on the command line and the certificate question is issued (Core(error): Failed to read response from stdin) ; if the 'yes' is not provided, the connection starts, the question is asked after which cause the application to froze.

**Comment by muzzol** [comment]
I'm facing this problem right now.

on my setup I use a kiosk user to connect fullscreen to a remote system without any intervention, so I need a way to accept certificate previously.

is there any progress with this issue?

**Comment by dronmaxman** [comment]
> 

To solve the problem, I compiled the rdesktop from the sources with the patch https://github.com/rdesktop/rdesktop/pull/364

Also, you can use xfreerdp client.

**Comment by oldlavender** [comment]
@dronmaxman it could be a temporary solution as long as @muzzol (or anyone else) is **pretty aware of the security issues it might cause**. As mentioned in #364, short options are not recommended at all for this kind of option. Typos happen and a user might end up mistaking some other option for it (specially with I - the uppercase "i", which might, in some fonts, be identical to an l - the lowercase "L", making it hard to notice the error), and thus disabling an important security step in contexts where it would not be the desired outcome.

It might cause the user **unnecessary exposure to a potential man in the middle attack**.

That said, you can use it temporarily and at your own risk, but this patch is 100% unadvised by us. 

I am working on a couple of new patches to support that in a proper way, the first one to enable long options, and as soon as it's done there will be an official way of doing that.

**Comment by muzzol** [comment]
I totally agree this can lead to security problems.

In my case offending cert is from our own servers, so I really need the option on my bash scripts.

Maybe a solution could be implementing a not easy parameter like **--no-check-certificate**
That's an actual example taken from wget and it doesn't have any short alternative, so its hard to type it by mistake.

Do you think that could be safe enough?

**Comment by dronmaxman** [comment]
@brunodOut Yes, thanks for your work. You can close this tiket.

I didn't find any other solutions to my problem. VMware-Horizont creates a tunnel on localhost and starts an rdesktop connection on localhost. Installing root certificates will not help in this case. The hostname of the terminal server (localhost)  and the CN in the certificate will all be different always.

**Comment by opoplawski** [comment]
Is it possible to just not validate the hostname part of the certificate?

**Comment by oldlavender** [comment]
@muzzol indeed, that would be safe and it's intended to be supported as soon as we implement long options. We had some historical issues with long options in some environments like Solaris, so I am investigating to decide a portable way of doing that. I think in 2-3 weeks it will be available for preview in the `next` branch, and then we can easily work on a `--no-check-certificate` option!

@dronmaxman I totally understand, and I am really sorry it took so long until some effort was put on a safer approach to this issue. But if you don't mind, I'd like to keep the ticket open for a couple of weeks yet, we will then have a proper option for that and you'll be able to use a safer and better alternative.

@opoplawski if I'm understanding what you mean, it's not possible and I don't think it would be a good idea. If the hostname you're connecting to and the one in the certificate don't match, there is no way to ensure the server you're connecting to is really the one you intended to connect or some other an attacker is using to get your password, for instance.

**Comment by angerangel** [comment]
Me too!

Windows server randomly change certificates, so we absolutely need the  **--no-check-certificate** option.  All our thin clients need it.

**Comment by cyguard2** [comment]
echo yes | rdesktop <whatever options>
worked for me.

**Comment by adrianodemoura** [comment]
for me ... i removed /home/MY_USER/.local/share/rdesktop, and traied again. works fine.

**Comment by arouene** [comment]
> echo yes | rdesktop worked for me.

Only works if you don't already have 
echo "password" | rdesktop -p -

The long option is the only option
